plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs

```diff
@@ -478,6 +479,14 @@ waitUntilDL = QCCM.waitUntilDL . toSlotNo
 assertModel :: String -> (QCCM.ModelState state -> Bool) -> DL state ()
 assertModel s p = QCCM.assertModel s (p . fmap coerce)
 
+-- IMPORTANT NOTE: the first argument needs to be unique among calls to `observeChain`
+-- as it is a proxy for the predicate.
+observeChain :: ContractModel state
+             => String
+             -> ((QCCM.SymToken -> CardanoAPI.AssetId) -> QCCM.ChainState -> Bool)
```

Comment:
The rest of the Plutus `Contract` `ContractModel` API is still using `AssetClass` instead of `CardanoAPI.AssetId`, right? I'm fine with either, but it should be consistent.

Reply:
Ah yes... It would be good if as we are transitioning the rest of plutus apps to cardano api we might as well transition the contract model. The current implementation is just a thin layer on top of a cardano api based implementation.